src/stories/_exec/__init__.py

```diff
@@ -0,0 +1,26 @@
+from ..exceptions import StoryDefinitionError
+from . import function, coroutine
+
+try:
+    from inspect import iscoroutinefunction
+except SyntaxError:
+    iscoroutinefunction = lambda x: False
+
+
+def get_executor(method, previous, cls_name):
+    if iscoroutinefunction(method):
+        executor = coroutine.execute
+    else:
+        executor = function.execute
+
+    if previous is not None and previous is not executor:
+        message = mixed_method_template.format(
+            cls=cls_name, method=method.__name__
+        )
+        raise StoryDefinitionError(message)
+    return executor
+
+
+mixed_method_template = """
+Class {cls} contains mixed method {method}'
```

Comment:
Should we show expected and actual executor in this message? Maybe error message like this will make more sense to the user:
```
Subscription.check_balance is a function but coroutine was expected.

Story method: Subscription.buy

You can not use function and coroutine methods together.
```
What do you think?